gulp/isotest.js

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ */
+"use strict";
+
+module.exports = (gulp, plugins, blueprint) => {
+    const rs = require("run-sequence").use(gulp);
+    const mocha = require("gulp-mocha");
+
+    blueprint.projectsWithBlock("isotest").forEach((project) => {
+        gulp.task(`isotest-${project.id}`, [`typescript-compile-${project.id}`], (done) => {
```

Comment:
This was not lost on me, but creating a task which will fail or give false results if you haven't compiled recently seems even worse. How about this: I create a second group of `isotest` task (perhaps `isotest-nobuild`?) which don't depend on the `typescript-compile` tasks. We use the 'nobuild' variant with CircleCI. The 'nobuild' variant can be run as easily as the 'buid' one, but the explicit naming makes the consequences clear, at least. 

For what it's worth, this is happening because `run-sequence` doesn't promote the dependencies to Gulp. If I had more time/motivation, `run-sequence` could be patched or re-written to correctly promote the dependencies, at which point Gulp would not re-run the tasks which had already been executed. Alternatively, the `typescript-compile`/ build tasks could perhaps be modified to test for freshness, though this seems rather hacky and suboptimal to me. Running tasks in sequence will be much easier when [Gulp 4 is released](https://github.com/gulpjs/gulp/milestone/1), but given that it's already 2 years in the works, who knows if it will ever be released.